maat-court-data-api/src/main/java/gov/uk/courtdata/assessment/impl/PassportAssessmentImpl.java

```diff
@@ -0,0 +1,75 @@
+        existingPassportAssessment.setPartnerSurname(passportAssessmentDTO.getPartnerSurname());
+        existingPassportAssessment.setPartnerNiNumber(passportAssessmentDTO.getPartnerNiNumber());
+        existingPassportAssessment.setPartnerDob(passportAssessmentDTO.getPartnerDob());
+        existingPassportAssessment.setIncomeSupport(passportAssessmentDTO.getIncomeSupport());
+        existingPassportAssessment.setJobSeekers(passportAssessmentDTO.getJobSeekers());
+        existingPassportAssessment.setStatePensionCredit(passportAssessmentDTO.getStatePensionCredit());
+        existingPassportAssessment.setPcobConfirmation(passportAssessmentDTO.getPcobConfirmation());
+        existingPassportAssessment.setResult(passportAssessmentDTO.getResult());
+        existingPassportAssessment.setDwpResult(passportAssessmentDTO.getDwpResult());
+        existingPassportAssessment.setPassportNote(passportAssessmentDTO.getPassportNote());
+        existingPassportAssessment.setUnder18HeardInYouthCourt(passportAssessmentDTO.getUnder18HeardInYouthCourt());
+        existingPassportAssessment.setUnder18HeardInMagsCourt(passportAssessmentDTO.getUnder18HeardInMagsCourt());
+        existingPassportAssessment.setLastSignOnDate(passportAssessmentDTO.getLastSignOnDate());
+        existingPassportAssessment.setEsa(passportAssessmentDTO.getEsa());
+        existingPassportAssessment.setUnder18FullEducation(passportAssessmentDTO.getUnder18FullEducation());
+        existingPassportAssessment.setUnder16(passportAssessmentDTO.getUnder16());
+        existingPassportAssessment.setBetween16And17(passportAssessmentDTO.getBetween16And17());
+        existingPassportAssessment.setPastStatus(passportAssessmentDTO.getPastStatus());
+        existingPassportAssessment.setWhoDWPChecked(passportAssessmentDTO.getWhoDWPChecked());
+        existingPassportAssessment.setDateCompleted(passportAssessmentDTO.getDateCompleted());
+        existingPassportAssessment.setUserModified(passportAssessmentDTO.getUserModified());
+        existingPassportAssessment.setDateModified(passportAssessmentDTO.getDateModified() != null ? passportAssessmentDTO.getDateModified() : LocalDateTime.now());
+        return passportAssessmentRepository.save(existingPassportAssessment);
+    }
+
+    public void delete(Integer passportAssessmentId) {
+        passportAssessmentRepository.deleteById(passportAssessmentId);
+    }
+
+    public PassportAssessmentEntity create(PassportAssessmentDTO passportAssessmentDTO) {
+        PassportAssessmentEntity passportAssessmentEntity = assessmentMapper.passportAssessmentDtoToPassportAssessmentEntity(passportAssessmentDTO);
+        return passportAssessmentRepository.save(passportAssessmentEntity);
+    }
+
+    public void setOldPassportAssessmentAsReplaced(PassportAssessmentDTO passportAssessment) {
+        passportAssessmentRepository.updatePreviousPassportAssessmentsAsReplaced(passportAssessment.getRepId(), passportAssessment.getId());
```

Comment:
This shouldn't match on the ID, the assessments we want to mark as replaced are those with a matching RepID but with a DIFFERENT id.